app/javascript/mastodon/features/interaction_modal/index.tsx

```diff
@@ -0,0 +1,585 @@
+      break;
+    case 'reblog':
+      icon = <Icon id='retweet' icon={RepeatIcon} />;
+      title = (
+        <FormattedMessage
+          id='interaction_modal.title.reblog'
+          defaultMessage="Boost {name}'s post"
+          values={{ name }}
+        />
+      );
+      action = (
+        <FormattedMessage
+          id='interaction_modal.action.reblog'
+          defaultMessage='boost'
+        />
+      );
+      break;
+    case 'favourite':
+      icon = <Icon id='star' icon={StarIcon} />;
+      title = (
+        <FormattedMessage
+          id='interaction_modal.title.favourite'
+          defaultMessage="Favorite {name}'s post"
+          values={{ name }}
+        />
+      );
+      action = (
+        <FormattedMessage
+          id='interaction_modal.action.favourite'
+          defaultMessage='favorite'
+        />
+      );
+      break;
+    case 'follow':
+      icon = <Icon id='user-plus' icon={PersonAddIcon} />;
+      title = (
+        <FormattedMessage
+          id='interaction_modal.title.follow'
+          defaultMessage='Follow {name}'
+          values={{ name }}
+        />
+      );
+      action = (
+        <FormattedMessage
+          id='interaction_modal.action.follow'
+          defaultMessage='follow'
+        />
+      );
+      break;
+    case 'vote':
+      icon = <Icon id='tasks' icon={InsertChartIcon} />;
+      title = (
+        <FormattedMessage
+          id='interaction_modal.title.vote'
+          defaultMessage="Vote in {name}'s poll"
+          values={{ name }}
+        />
+      );
+      action = (
+        <FormattedMessage
+          id='interaction_modal.action.vote'
+          defaultMessage='vote'
+        />
+      );
+      break;
+  }
+
+  let signupButton;
+
+  if (sso_redirect) {
+    signupButton = (
+      <a href={sso_redirect} data-method='post' className='link-button'>
+        <FormattedMessage
+          id='sign_in_banner.create_account'
+          defaultMessage='Create account'
+        />
+      </a>
+    );
+  } else if (registrationsOpen) {
+    signupButton = (
+      <a href={signupUrl} className='link-button'>
+        <FormattedMessage
+          id='sign_in_banner.create_account'
+          defaultMessage='Create account'
+        />
+      </a>
+    );
+  } else {
+    signupButton = (
+      <button className='link-button' onClick={handleSignupClick}>
+        <FormattedMessage
+          id='sign_in_banner.create_account'
+          defaultMessage='Create account'
+        />
+      </button>
+    );
+  }
+
+  return (
+    <div className='modal-root__modal interaction-modal'>
+      <div className='interaction-modal__lead'>
+        <h3>
+          <span className='interaction-modal__icon'>{icon}</span> {title}
+        </h3>
+        <p>
+          <FormattedMessage
+            id='interaction_modal.to_continue_prompt'
+            defaultMessage='To continue, you need to {action} from your account.'
+            values={{ action }}
+          />
```

Comment:
This is *probably* fine, but I'd be careful with this. It's usually way better to have full sentences in translatable strings, even if similar strings end up repeating themselves. This gives translators more context and helps with languages which have a different grammatical structure.